tests/Oryx.BuildImage.Tests/Python/PythonSampleAppsTest.cs

```diff
@@ -400,6 +400,90 @@ public void GeneratesScript_AndBuilds_WhenExplicitLanguageAndVersion_AreProvided
                 result.GetDebugInfo());
         }
 
+        // This is to test if we can build an app when there is no requirement.txt
+        // but setup.py is provided at root level
+        [Fact]
+        public void GeneratesScript_AndBuilds_WhenSetupDotPy_File_isProvided()
+        {
+            // Arrange
+            var appName = "flask-setup-py-app";
+            var volume = CreateSampleAppVolume(appName);
+            var appDir = volume.ContainerDir;
+            var appOutputDir = $"{appDir}/output";
+            var manifestFile = $"{appOutputDir}/{FilePaths.BuildManifestFileName}";
+            var script = new ShellScriptBuilder()
+                .AddBuildCommand(
+                $"{appDir} -o {appOutputDir} --platform python --platform-version {PythonVersions.Python36Version}")
+                .AddCommand($"cat {manifestFile}")
+                .ToString();
+
+            // Act
+            var result = _dockerCli.Run(new DockerRunArguments
+            {
+                ImageId = Settings.BuildImageName,
+                EnvironmentVariables = new List<EnvironmentVariable> { CreateAppNameEnvVar(appName) },
+                Volumes = new List<DockerVolume> { volume },
+                CommandToExecuteOnRun = "/bin/bash",
+                CommandArguments = new[] { "-c", script }
+            });
+
+            // Assert
+            RunAsserts(
+                () =>
+                {
+                    Assert.True(result.IsSuccess);
+                    Assert.Contains(
+                        $"Python Version: /opt/python/{PythonVersions.Python36Version}/bin/python3",
+                        result.StdOut);
+                    Assert.Contains(
+                       $"{ManifestFilePropertyKeys.PythonVersion}=\"{PythonVersions.Python36Version}\"",
+                       result.StdOut);
+                },
+                result.GetDebugInfo());
+        }
+
+        // This is to test if we can build an app when both the files requirement.txt
+        // and setup.py are provided, we tend to prioritize the root level requirement.txt
+        [Fact]
+        public void GeneratesScript_AndBuilds_With_Both_Files_areProvided()
+        {
+            // Arrange
+            var appName = "flask-app";
```

Comment:
The test is talking about 'both' files being provided but this app only has `requirements.txt` file and not `setup.py`. Did you want to use `flask-setup-py-requirement-txt` app instead?

Reply:
yes .. good catch … thanks !